app/src/main/java/com/example/bottomnav/MainActivity.java

```diff
@@ -1,17 +1,32 @@
 package com.example.bottomnav;
 
+import android.Manifest;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
 import android.os.Bundle;
-import com.google.android.material.bottomnavigation.BottomNavigationView;
+import android.os.IBinder;
+import android.util.Log;
+import android.widget.ArrayAdapter;
+import android.widget.ExpandableListView;
+import android.widget.ListView;
+
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
 import androidx.navigation.NavController;
 import androidx.navigation.Navigation;
 import androidx.navigation.ui.AppBarConfiguration;
 import androidx.navigation.ui.NavigationUI;
+
+import com.example.bottomnav.bluetoothlegatt.DeviceControlActivity;
+import com.example.bottomnav.databinding.ActivityMainBinding;
+import com.example.bottomnav.BluetoothLeService;
```

Comment:
Can you delete things instead of commenting them out?

Commenting things out is fine for testing and development, but for maintaining a clean codebase committing commented out code is discouraged.